guides/building-a-rest-api-spring-boot-vs-micronaut-post/building-a-rest-api-spring-boot-vs-micronaut-post.adoc

```diff
@@ -0,0 +1,92 @@
+common:header-top.adoc[]
+
+== Sample Project
+
+You can link:@sourceDir@.zip[download a sample application] with the code examples in this article.
+
+== Introduction
+
+This guide compares how to write a POST endpoint backed by a persistence layer in a Micronaut Framework and Spring Boot applications.
+
+The Spring Boot application uses https://spring.io/projects/spring-data[Spring Data] and https://www.h2database.com/html/main.html[H2]. The Micronaut application uses https://micronaut-projects.github.io/micronaut-data/snapshot/guide/[Micronaut Data] and H2.
+Micronaut Data is a database access toolkit that uses Ahead of Time (AoT) compilation to pre-compute queries for repository interfaces that are then executed by a thin, lightweight runtime layer.
+
+This guide is the fourth tutorial of https://guides.micronaut.io/latest/tag-building_a_rest_api.html[Building a Rest API] - a series of tutorials comparing how to develop a REST API with Micronaut Framework and Spring Boot.
+
+== Controller
+
+The controller receives a POST request with a JSON body, the repository persists the `SaasSubscription`, and then it returns
+a 201 response with a `Location` HTTP Header.
+
+As you will see, both controllers are almost identical, with similar annotations, as we discussed in the https://guides.micronaut.io/latest/guides/building-a-rest-api-spring-boot-vs-micronaut-implemeting-get.html[second guide] of this series.
+
+The new APIs used in the following controllers are similar:
+
+[.left-stripes-even,cols="1h,2*"]
+.Comparison between Spring Boot and Micronaut Framework
+|===
+| | Spring Boot | Micronaut
+
+
+| Identify a method as a POST endpoint
+| `@PostMapping`
+| `@Post`
+
+| Read and deserialize the request body to a method parameter
+| `@RequestBody`
+| `@Body`
+
+| Fluid ApI to build a URI
+| `UriComponentsBuilder`
+| `UriBuilder`
+|===
+
+=== Spring Boot Controller
+
+source:SaasSubscriptionPostController[app=springboot]
+
+callout:spring-boot-rest-controller[1]
+callout:spring-boot-request-mapping[2]
+callout:constructor-di[number=3,arg0=SaasSubscriptionRepository]
+callout:spring-boot-post-mapping[number=4,arg0=findById,arg1=/subscriptions/{id}]
```

Comment:
This renders

> The @PostMapping annotation maps the findById method to an HTTP POST request on /subscriptions/{id}.

Which is not what this method is doing...

Should it be:

> The @PostMapping annotation maps this method to a POST request.

?

Reply:
resolved via https://github.com/micronaut-projects/micronaut-guides/pull/1457/commits/eece9ef15db2941179d5aeb6fae45aed3701ab58